### PR TITLE
Fix/tags in types

### PR DESCRIPTION
**Pull request by MateuszKubuszok**
WIP

### CODE REVIEW

File: chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala [hunk 1]
```diff
@@ -8,161 +8,162 @@ import scala.collection.generic.CanBuildFrom
...
+    (src: T, _: Modifiers) => tag[U](src)
+
+  implicit def fromTaggedTransformer[T, U, Modifiers <: HList]
+  : DerivedTransformer[T @@ U, T @@ U, T, Modifiers] =
+    (src: T @@ U, _: Modifiers) => src
+
+  implicit def passTaggedTransformer[T, U, Modifiers <: HList]
+  : DerivedTransformer[T @@ U, T @@ U, T @@ U, Modifiers] =
+    (src: T @@ U, _: Modifiers) => src
+}
```
Comment (MateuszKubuszok) [inline chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala]: probably unnecessary, remove before finishing task

File: chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala [hunk 2]
```diff
@@ -1,168 +1,130 @@
...
-   tailTransformer: DerivedTransformer[TailFromG @@ From, ToG, Modifiers])
-  : DerivedTransformer[@@[FieldType[Label, HeadT] :+: TailFromG, From], ToG, Modifiers] =
-    (src: @@[FieldType[Label, HeadT] :+: TailFromG, From], modifiers: Modifiers) =>
-      (src: FieldType[Label, HeadT] :+: TailFromG) match {
-        case Inl(hd) => cip.provide(hd, modifiers)
-        case Inr(tl) => tailTransformer.transform(tag[From](tl), modifiers)
-      }
-
-  implicit def gen[From, To, FromG, ToG, Modifiers <: HList]
+  implicit def withIntermediateProductRepr[From, To, FromG <: HList, ToG <: HList, Modifiers <: HList]
```
Comment (krzemin) [inline chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala]: genProduct?

File: chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala [hunk 3]
```diff
@@ -1,168 +1,130 @@
...
     (src: From, modifiers: Modifiers) =>
-      toLG.from(genTransformer.value.transform(tag[From](fromLG.to(src)), modifiers))
+      toLG.from(genTransformer.value.transform(fromLG.to(src), modifiers))
   }
-}
-
-trait LowPriorityGenericInstances {
 
-  implicit def gen2[From, To, FromG, ToG, Modifiers <: HList]
+  implicit def withIntermediateCoproductRepr[From, To, FromG <: Coproduct, ToG <: Coproduct, Modifiers <: HList]
```
Comment (krzemin) [inline chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala]: genCoproduct?

File: chimney/src/main/scala/io/scalaland/chimney/DerivedCoproductTransformer.scala [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import shapeless.labelled.FieldType
+
+trait DerivedCoproductTransformer[From, FromG <: Coproduct, ToG <: Coproduct, Modifiers <: HList] {
+
+  def transform(src: FromG, modifiers: Modifiers): ToG
+}
+
+object DerivedCoproductTransformer extends CoproductInstances {
+
+  def apply[From, FromG <: Coproduct, ToG <: Coproduct, Modifiers <: HList](implicit dt: DerivedCoproductTransformer[From, FromG, ToG, Modifiers]): DerivedCoproductTransformer[From, FromG, ToG, Modifiers] = dt
```
Comment (krzemin) [inline chimney/src/main/scala/io/scalaland/chimney/DerivedCoproductTransformer.scala]: not so usable in this form, but we can keep it

**Pull request merged**